airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/CdcTargetPosition.java

```diff
@@ -74,7 +74,7 @@ default boolean isEventAheadOffset(final Map<String, String> offset, final Chang
    * @return Returns `true` if both offsets are at the same position. Otherwise, it returns `false`
    */
   default boolean isSameOffset(final Map<String, String> offsetA, final Map<String, String> offsetB) {
-    return true;
+    return false;
```

Comment:
Why this change?

Reply:
This came with Marius' heartbeat change
@postamar Why did the default change?

Reply:
A default value of `true` makes checkpointing impossible, see implementation of `DebeziumStateDecoratingIterator::computeNext`: `checkpointOffsetToSend` will never be assigned a value. This is despite the fact that the purpose of the test involving `isSameOffset`, which guards this assignment, appears to be to avoid emitting more than one state message with the same state.